db-tests/src/test/java/gov/nasa/jpl/aerie/database/ExternalEventTests.java

```diff
@@ -0,0 +1,1922 @@
+   *    derivation_group_comp view, and deletion.
+   * Aside from the derivation_group_comp view test, these tests are not rigorous (more rigorous tests follow in the
+   *    ConstraintsTests class).
+   */
+  @Nested
+  @TestInstance(TestInstance.Lifecycle.PER_CLASS)
+  class BasicTests {
+    /**
+     * This first test is to verify that a simple upload works, regardless of if its output is correct.
+     * As we do not upload a file we just run the queries that would be run if files were uploaded,
+     * namely tests 01-04 from SSMO-MPS/mission-data-sandbox/derivation_test.
+     */
+    @Test
+    void uploadWithoutError() {
+      assertDoesNotThrow(() -> upload_source(dg));
+    }
+
+
+    /**
+     * This test uploads data (tests 01-04 from SSMO-MPS/mission-data-sandbox/derivation_test) and then confirms derived
+     * events associated with the upload perform as expected. All desired keys should be present.
+     * This is an extension of uploadWithoutError.
+     */
+    @Test
+    void basicDerivedEvents() throws SQLException {
+      // upload all source data
+      assertDoesNotThrow(() -> upload_source(dg));
+      try(final var statement = connection.createStatement()) {
+
+        // check that derived events in our prewritten case has the correct keys
+        assertDoesNotThrow(() -> {
+          final var res = statement.executeQuery(
+              // language=sql
+              """
+              SELECT * FROM merlin.derived_events ORDER BY start_time;
+              """
+          );
+
+          final String[] expected_keys = {"1", "9", "3", "5", "6", "2", "8"};
+          compareLists(expected_keys, res, "event_key");
+        });
+      }
+    }
+
+    /**
+     * This test uploads data (tests 01-04 from SSMO-MPS/mission-data-sandbox/derivation_test) and then confirms
+     * derivation_group_comp view associated with upload performs as expected.
+     * This test alone is rigorous enough for derivation_group_comp view, and is an extension of uploadWithoutError.
+     */
+    @Test
+    void derivationGroupComp() throws SQLException {
+      // upload all source data
+      assertDoesNotThrow(() -> upload_source(dg));
+      try(final var statement = connection.createStatement()) {
+
+        // check that derivation_group_comp has 1 entry, with 4 sources and 7 events
+        assertDoesNotThrow(() -> {
+          final var res = statement.executeQuery(
+              // language=sql
+              """
+              SELECT * FROM ui.derivation_group_comp;
+              """
+          );
+
+          // we expect only 1 result
+          assertTrue(res.next());
+
+          // res.getArray() doesn't give you a String[] array - have to do some manipulation to get its length as a result
+          System.out.println(res.getArray("sources").toString());
+          assertEquals(4, res.getArray("sources").toString().split("\",\"").length);
+          assertEquals(7, res.getInt("derived_total"));
```

Comment:
The way this is handled in our other tests is that we have a method that runs the `select` statement and processes the output into a List of the appropriate record. Please use `TagsTests.getAllTags()` as reference.

Reply:
We don't need to check source properties, just that this list has 4 elements, for which this split should be sufficient. There isn't enough data included in these entries to even make a proper `ExternalSource` object from them, unless we create some extra type which would only see use here. What are you suggesting we do here?

Reply:
I saw the later comment. Your recommendation is we make a type specific to `derived_events.sql` selection results and work with that type?

Reply:
I suggest that you refine the test so it is actually checking the contents of these arrays rather than just saying "oh there's four elements that's good enough".

Reply:
Addressed in [update derivationGroupComp test](https://github.com/NASA-AMMOS/aerie/commit/05e961e00cbd3903dafe3fcd003ae0886a65c16d).